packages/executor/src/services/EventsService/versions/0.0.7.ts

```diff
@@ -29,67 +126,89 @@ export class EntryPointV7EventsService {
   }
 
   initEventListener(): void {
-    this.contract.on(
-      this.contract.filters.UserOperationEvent(),
-      async (...args) => {
+    this.publicClient.watchContractEvent({
+      abi: EntryPoint__factory.abi,
+      eventName: "UserOperationEvent",
+      address: this.entryPoint,
+      onLogs: async (args) => {
         const ev = args[args.length - 1];
-        await this.handleEvent(ev as ParsedEventType);
+        await this.handleUserOperationEvent(ev);
       }
-    );
-  }
+    });
 
-  onUserOperationEvent(callback: TypedListener<UserOperationEventEvent>): void {
-    this.contract.on(this.contract.filters.UserOperationEvent(), callback);
-  }
+    this.publicClient.watchContractEvent({
+      abi: EntryPoint__factory.abi,
+      address: this.entryPoint,
+      eventName: "AccountDeployed",
+      onLogs: async (args) => {
+        const ev = args[args.length - 1];
+        await this.handleAccountDeployedEvent(ev);
+      }
+    });
 
-  offUserOperationEvent(
-    callback: TypedListener<UserOperationEventEvent>
-  ): void {
-    this.contract.off(this.contract.filters.UserOperationEvent(), callback);
+    this.publicClient.watchContractEvent({
+      abi: EntryPoint__factory.abi,
+      address: this.entryPoint,
+      eventName: "SignatureAggregatorChanged",
+      onLogs: async (args) => {
+        const ev = args[args.length - 1];
+        await this.handleAggregatorChangedEvent(ev);
+      }
+    });
   }
 
+  // onUserOperationEvent(callback: TypedListener<UserOperationEventEvent>): void {
+  //   this.contract.on(this.contract.filters.UserOperationEvent(), callback);
+  // }
+
+  // offUserOperationEvent(
+  //   callback: TypedListener<UserOperationEventEvent>
+  // ): void {
+  //   this.contract.off(this.contract.filters.UserOperationEvent(), callback);
+  // }
+
   /**
    * manually handle all new events since last run
    */
-  async handlePastEvents(): Promise<void> {
-    await this.fetchLastBlockPerEntryPoints();
-    const events = await this.contract.queryFilter(
-      { address: this.entryPoint },
-      this.lastBlock
-    );
-    for (const ev of events) {
-      await this.handleEvent(ev);
-    }
-    if (events.length > 0) {
-      const lastEvent = events[events.length - 1];
-      const blockNum = lastEvent!.blockNumber;
-      if (!this.lastBlock || Number(this.lastBlock) < blockNum) {
-        this.lastBlock = blockNum;
-      }
-    }
-    await this.saveLastBlockPerEntryPoints();
-  }
+  // async handlePastEvents(): Promise<void> {
```

Comment:
Please delete unused code completely. We can always restore from history if required.